### PR TITLE
Audio component

### PR DESCRIPTION
**Pull request by cjharkins**
- Connects Audio uploader to AWS

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -38,7 +38,8 @@ import defaultMessages from './translations/en.json'
 
 // Step 3:
 // If you are using a non-english locale, point `messagesInLocale` to correct .json file
-import messagesInLocale from './translations/en.json' // TODO
+import messagesInLocale from './translations/en.json'
+import { ServiceTypeProvider } from './context/ServiceTypeProvider' // TODO
```
Comment (embryCODE) [inline src/app.js:42]: Did you do the TODO? Should we make a ticket for it?

File: src/components/EditListingWizard/EditListingWizard.js [hunk 2]
```diff
@@ -82,18 +89,33 @@ const tabLabel = (intl, tab) => {
  * @return true if tab / step is completed.
  */
 const tabCompleted = (tab, listing) => {
-  const { availabilityPlan, description, price, title, publicData } = listing.attributes
+  const { availabilityPlan, price, title, publicData } = listing.attributes // where does this come from?
```
Comment (embryCODE) [inline src/components/EditListingWizard/EditListingWizard.js:92]: Did we find this out?

File: src/components/EditListingWizard/EditListingWizard.js [hunk 3]
```diff
@@ -82,18 +89,33 @@ const tabLabel = (intl, tab) => {
...
+      const { whyAreYouTheRightFit, primaryGenres, experience } = publicData
+      return !!whyAreYouTheRightFit && !!primaryGenres && !!experience
+    case ABOUTTHISSERVICE:
+      const { averageTurnAroundTime, explainMore } = publicData
+      return !!averageTurnAroundTime && !!explainMore
+    case PAYMENT:
+      const { payPalEmail, phoneNumber } = publicData
+      return !!payPalEmail && !!phoneNumber
+    case TERMSOFUSE:
+    // TODO: Write validation
```
Comment (embryCODE) [inline src/components/EditListingWizard/EditListingWizard.js:113]: Make a Notion ticket for this so it doesn't get lost?

File: src/components/EditListingWizard/EditListingWizardTab.js [hunk 4]
```diff
@@ -207,6 +281,7 @@ const EditListingWizardTab = (props) => {
           {...panelProps(PRICING)}
           submitButtonText={intl.formatMessage({ id: submitButtonTranslationKey })}
           onSubmit={(values) => {
+            console.log(values)
```
Comment (embryCODE) [inline src/components/EditListingWizard/EditListingWizardTab.js:284]: 🔥

File: src/forms/EditListingAudioForm/EditListingAudioForm.css [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    clear: both;
+  }
+
+  @media (--viewportLarge) {
+    width: var(--EditListingAudioForm_imageWidthDesktop);
+    margin: 0 0 24px 0;
+  }
+}
+
+/* Firefox doesn't support image aspect ratio inside flexbox */
```
Comment (embryCODE) [inline src/forms/EditListingAudioForm/EditListingAudioForm.css:64]: This reminds me. We'll want a ticket to check this on Firefox, Edge, Chrome, and Safari. Good news is that Edge is Chromium now, so basically Chrome with MS skinning. If Andrew makes me support IE or old Edge Ima charge him more. Haha.

File: src/forms/EditListingAudioForm/EditListingAudioForm.example.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-console */
+import EditListingAudioForm from './EditListingAudioForm'
+
+export const Empty = {
+  component: EditListingAudioForm,
+  props: {
+    initialValues: { country: 'US', images: [] },
+    stripeConnected: false,
+    onImageUpload: (values) => {
+      console.log(`onImageUpload with id (${values.id}) and file name (${values.file.name})`)
```
Comment (embryCODE) [inline src/forms/EditListingAudioForm/EditListingAudioForm.example.js:10]: These logs can stay if you need them but if things are working well we will eventually remove, of course.

File: src/forms/EditListingAudioForm/EditListingAudioForm.js [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+const ACCEPT_AUDIO = 'audio/*'
+
+export class EditListingAudioFormComponent extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { audioUploadRequested: false, currentFileName: '', currentFileProgress: 0  }
+    this.uploader = null
+    this.handleUpload = this.handleUpload.bind(this)
+    this.handleUploadProgress = this.handleUploadProgress.bind(this)
+    // this.onImageUploadHandler = this.onImageUploadHandler.bind(this)
```
Comment (embryCODE) [inline src/forms/EditListingAudioForm/EditListingAudioForm.js:24]: 🔥

File: src/forms/EditListingAudioForm/EditListingAudioForm.js [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+      this.setState({currentFileProgress: value})
+    }
+  }
+
+  handleUpload(fileName) {
+    this.setState({ audioUploadRequested: true })
+    this.submittedAudio = [...this.submittedAudio, fileName]
+  }
+
+  onAudioUploadHandler(file) {
```
Comment (embryCODE) [inline src/forms/EditListingAudioForm/EditListingAudioForm.js:41]: 🔥

File: src/forms/EditListingAudioForm/EditListingAudioForm.js [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+            invalid || disabled || submitInProgress || audioUploadRequested || ready
+
+          const classes = classNames(css.root, className)
+
+          const listAudio = this.submittedAudio.length > 0 ?
+            (<ul>
+              { this.submittedAudio.map((fileName, index) =>
+                <li key={fileName + index}>
+                  {fileName}
+                  { this.state.currentFileName === fileName ? <div style={{width: 200, height: 15, border: '1px solid red'}}>
```
Comment (embryCODE) [inline src/forms/EditListingAudioForm/EditListingAudioForm.js:134]: This is fine but if it gets any bigger let's make styled components

File: src/forms/EditListingAudioForm/EditListingAudioForm.js [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+
+              { listAudio }
+
+              <ReactS3Uploader
+                signingUrl="/s3/sign"
+                signingUrlMethod="GET"
+                accept={ACCEPT_AUDIO}
+                // preprocess={(thing)=> console.log({thing})}
+                onSignedUrl={(res) => {
+                  this.handleUploadProgress({ type: 'fileName', fileName: res.filename })
```
Comment (embryCODE) [inline src/forms/EditListingAudioForm/EditListingAudioForm.js:163]: Remind me over whiskey and I'll show you a trick for this kind of thing using closures. 🥃

**Review by embryCODE — APPROVED** [review]
Couple of comments. Feel free to merge afterwards.

File: src/components/EditListingWizard/EditListingWizard.js (hunk 2 above)
Comment (embryCODE) [reply src/components/EditListingWizard/EditListingWizard.js:92]: Nevermind. Just saw your other PR comments.

File: src/app.js (hunk 1 above)
Comment (cjharkins) [reply src/app.js:42]: No need for TODO tickets, just a reminder for me, removing them

File: src/forms/EditListingAudioForm/EditListingAudioForm.example.js (hunk 6 above)
Comment (cjharkins) [reply src/forms/EditListingAudioForm/EditListingAudioForm.example.js:10]: This is existing code for the examples

**Pull request merged**